capa/engine.py

```diff
@@ -105,10 +104,26 @@ def __nonzero__(self):
         return self.success
 
 
+class Description(object):
+    def __init__(self, value=None):
+        super(Description, self).__init__()
+        self.name = self.__class__.__name__
+        self.value = value
+
+    def __str__(self):
+        return "%s" % (self.value)
+
+    def __repr__(self):
+        return str(self)
+
+
 class And(Statement):
     """match if all of the children evaluate to True."""
 
-    def __init__(self, children, description=None):
+    def __init__(self, children):
+        description = get_statement_description(children)
+        if description:
+            children.remove(description)
```

Comment:
how about making this routine a method on `class Statement`? keeps the code more closely associated with the related classes.

then could make it a mutator, like `pop_description(...)` which both finds the description and removes it from `children`, returning it.

which makes me wonder if it can work directly on `self.children` with no explicit parameters. in which case, it could be called directly from the `Statement.__init__()`. then we wouldn't have to change code in all the descendent classes?

Reply:
are there any statement types that don't support `description`? maybe `regex`?